### ISSUE TITLE
`Naxsi-Config : Incorrect line MainRule id:1500` after upgrading to os-nginx 1.32.1_1

### ISSUE THREAD

**Issue by mreiche** [opnsense/plugins#3480]
**Important notices**
- [✓] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [✓] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [✓] The title contains the plugin to which this issue belongs

**Describe the bug**
I've upgraded the Opnsense version of the secondary WAF from 21.1.9 to 23.1.10_1 for upgrade testing purposes. There weren't any errors during the update. When I synchronized the configuration from primary, I noticed that the Nginx service was not starting because of the errors below.

Working packages/plugins for OpnSense 21.1.9
- os-nginx 1.28
- Nginx 1.20.2_9,2

Not working packages/plugins for OpnSense 23.1.10_1
- os-nginx 1.32.1_1
- nginx 1.24.0_6,3

Removing rule 1500 did not help at all. I had to uncomment the line in `nginx.conf`, but then it stops at another incorrect rule.

**Relevant log files**

```
nginx: configuration file /usr/local/etc/nginx/nginx.conf test failed |  
nginx: [emerg] Naxsi-Config : Incorrect line MainRule id:1500 (/usr/obj/usr/ports/www/nginx/work/naxsi-29793dc/naxsi_src/naxsi_skeleton.c/973)... in /usr/local/etc/nginx/nginx.conf:33
nginx: [emerg] Naxsi-Config : Incorrect line BasicRule id:2003 (/usr/obj/usr/ports/www/nginx/work/naxsi-29793dc/naxsi_src/naxsi_skeleton.c/555)... in /usr/local/etc/nginx/nginx.conf:332
```

**Environment** 
OPNsense 23.1.10_1 (amd64, OpenSSL).

**Comment by mreiche** [comment]
Update: We've upgraded step-by-step and the last working version is:

* OpnSense: 23.1_6
* nginx 1.22.0_7,3
* os-nginx: 1.31

**Comment by mreiche** [comment]
May be related to https://github.com/opnsense/plugins/issues/3481

**Comment by fichtner** [comment]
https://forum.opnsense.org/index.php?topic=34546.msg167432#msg167432 states:

> after the hotfix, I have narrowed it down. When you disable the naxsi rules:
> rules SQL Injections 1000-1099 and File Uploads 1500-1600
> Nginx starts and works. Hope someone else can see what the "problem" is with naxsi / these rules with the latest nginx

I'm leaving this for more capable hands. Not sure what nginx's goal is here.

**Comment by mreiche** [comment]
It seems that regular expressions are not working anymore (https://github.com/nbs-system/naxsi/issues/635).

**Comment by fichtner** [comment]
The FreeBSD port is still using https://github.com/nbs-system/naxsi so I'm going to try their latest commit first... @wargio seems involved

**Comment by wargio** [comment]
The original repo has been abandoned by the company and as last developer i gave them 1y of support in case they wanted to continue the development, but nothing has happen therefore the forking.
1.4 has many fixes (especially for the latest versions of nginx) and improvements.

**Comment by fichtner** [comment]
@wargio thanks! I'll try to get FreeBSD ports to move to your repo but I'd like to do that in small steps as we also have a production release affected by this and I can't do much QA in production for problems created in FreeBSD ports which we just happen to use.

**Comment by fichtner** [comment]
@mreiche can you try this build?

    # opnsense-revert -z nginx

**Comment by fichtner** [comment]
@mreiche now I have a working build for @wargio's 1.4:

    # opnsense-revert -z nginx

To avoid confusion the temporary version is 1.24.0_8,3 -- FreeBSD's version is a bit funky ;)


Cheers,
Franco

**Comment by mreiche** [comment]
Fixed in OpnSense 23.1.11 and os-nginx 1.32.1_2
